tox.ini

```diff
@@ -2,83 +2,128 @@
     python -c "import psutil; proc_name = 'Ans.Dpf.Grpc'; nb_procs = len([proc.kill() for proc in psutil.process_iter() if proc_name in proc.name()]); \
     print(f'Killed \{nb_procs} \{proc_name} processes.')"
 
-    # Organize test files
+[testenv:pretest]
+description = Environment to organize test files prior to testing
+
+skip_install = True
+
+deps =
+
+commands_pre =
+
+commands = 
     python -c "\
     import os, shutil; \
     test_data=['test_launcher','test_server','test_local_server','test_multi_server','test_workflow','test_remote_workflow','test_remote_operator','test_service','test_custom_type_field']; \
     [(os.makedirs(d, exist_ok=True), shutil.copy('tests/conftest.py', d), shutil.copy(f'tests/\{d}.py', d) if os.path.exists(f'tests/\{d}.py') else None) for d in test_data]; \
     [os.remove(f'tests/\{d}.py') for d in test_data if os.path.exists(f'tests/\{d}.py')]"
 
 [testenv:posttest]
-description = Environment to kill servers and revert test files to original state after testing
+description = Environment to revert test files to original state after testing
+
+depends = pretest, test-{api,launcher,server,local_server,multi_server,remote_workflow,remote_operator,workflow,service,api_entry,custom_type_field,operators}
 
-depends = pretest, test-{api,launcher,server,local_server,multi_server,remote_workflow,remote_operator,workflow,service,operators}
+skip_install = True
 
 deps =
-    psutil
 
-skip_install = True
+commands_pre =
 
 commands = 
-    # Revert project layout to previous state
     python -c "\
     import os, shutil; \
     test_data=['test_launcher','test_server','test_local_server','test_multi_server','test_workflow','test_remote_workflow','test_remote_operator','test_service', 'test_custom_type_field']; \ 
     [shutil.move(f'\{d}/\{d}.py', f'tests/\{d}.py') for d in test_data if os.path.exists(f'\{d}/\{d}.py')]; \
     [shutil.rmtree(d) for d in test_data if os.path.exists(d)]"
 
-    # Clear any running servers that may be locking resources
-    python -c "import psutil; proc_name = 'Ans.Dpf.Grpc'; nb_procs = len([proc.kill() for proc in psutil.process_iter() if proc_name in proc.name()]); \
-    print(f'Killed \{nb_procs} \{proc_name} processes.')"
-
-[testenv:test-{api,launcher,server,local_server,multi_server,remote_workflow,remote_operator,workflow,service,operators}]
+[testenv:test-{api,launcher,server,local_server,multi_server,remote_workflow,remote_operator,workflow,service,api_entry,custom_type_field,operators}]
 description = Environment where project testing configuration is defined
 
 depends = pretest
 
 setenv =
     # Pytest extra arguments
-    COVERAGE = --cov=ansys.dpf.core --cov-report=xml --cov-report=html --log-level=ERROR --cov-append
+    COVERAGE = --cov=ansys.dpf.core --log-level=ERROR
     RERUNS = --reruns=2 --reruns-delay=1
     DEBUG = -v -s --durations=10 --durations-min=1.0
+    COVERAGE_FILE = {work_dir}/.cov/.coverage.{env_name}
```

Comment:
Excellent! So using `{env_name}` allows us to create a cov file with the name of the environment that created it, right? This is great! It avoids passing environment variables.